pkg/nsx/services/common/store.go

```diff
@@ -62,6 +62,37 @@ func DecrementPageSize(pageSize *int64) {
 	}
 }
 
+func (resourceStore *ResourceStore) ListIndexFuncValues(key string) sets.String {
+	values := sets.NewString()
+	entities := resourceStore.Indexer.ListIndexFuncValues(key)
+	for _, entity := range entities {
+		values.Insert(entity)
+	}
+	return values
+}
+
+// GetByKey is the method to get the resource by key, it is used by the subclass
+// to assert it to the specific type.
+func (resourceStore *ResourceStore) GetByKey(key string) interface{} {
+	res, exists, err := resourceStore.Indexer.GetByKey(key)
+	if err != nil {
+		log.Error(err, "failed to get obj by key", "key", key)
+	} else if exists {
+		return res
+	}
+	return nil
+}
+
+// GetByIndex is the method to get the resource list by index, it is used by the subclass
+// to assert it to the specific type.
```

Comment:
to convert it to the specific type. -> maybe better?

Reply:
ok